src/roswire/common/source.py

```diff
@@ -29,100 +32,70 @@ class SourceLanguage(enum.Enum):
-            opts, args = cmake_argparse(
-                args,
-                {"PARENT_SCOPE": "-", "FORCE": "-", "CACHE": "*"}
-            )
-            cmake_env[args[0]] = ";".join(args[1:])
-        elif cmd == "unset":
-            opts, args = cmake_argparse(args, {"CACHE": "-"})
-            del cmake_env[args[0]]
-        elif cmd == "add_executable":
-            name = args[0]
-            sources: t.List[str] = []
-            for _token_type, token_val in arg_tokens[1:]:
-                if not token_val.startswith("$"):
-                    sources.append(token_val)
-                else:
-                    matches = re.match(r'\${(.*)}', token_val)
-                    if matches and matches.group(1) in cmake_env:
-                        sources.extend(cmake_env[matches.group(1)].split(";"))
-            src_info.add(NodeSourceInfo(name, SourceLanguage.CXX, sources))
-        elif cmd == "catkin_install_python":
-            opts, args = cmake_argparse(
-                args,
-                {"PROGRAMS": "*", "DESTINATION": "*"}
-            )
-            if 'PROGRAMS' in opts:
-                program_opts = opts['PROGRAMS']
-                for i in range(len(program_opts)):
-                    # http://docs.ros.org/en/jade/api/catkin/html/howto/format2/installing_python.html  # noqa: E501
-                    # Convention is that ros python nodes are in nodes/
-                    # directory. All others are in scripts/. So just
-                    # include python installs that are in nodes/
-                    program = program_opts[i]
-                    if program.startswith("nodes/"):
-                        name = Path(program[0]).stem
-                        sources = [program]
-                        src_info.add(
-                            NodeSourceInfo(
-                                name,
-                                SourceLanguage.PYTHON,
-                                sources))
-            else:
-                raise ValueError(
-                    "PROGRAMS not specified in catkin_install_python"
-                )
-    return src_info
+
+@attr.s(auto_attribs=True, slots=True)
+class NodeletExecutableInfo(ExecutableInfo):
+    entrypoint: str
+
+    def to_dict(self) -> t.Dict[str, t.Any]:
+        d = super().to_dict()
+        d['entrypoint'] = self.entrypoint
+        return d
+
+    @classmethod
+    def from_dict(cls, info: t.Dict[str, t.Any]) -> 'NodeletExecutableInfo':
+        return NodeletExecutableInfo(info["name"],
+                                     SourceLanguage(info["language"]),
+                                     ExecutableKind(info["kind"]),
+                                     set(info["sources"]),
+                                     set(info["path_restrictions"]),
+                                     info['entrypoint'])
+
+
+@attr.s(auto_attribs=True, slots=True, frozen=True)
+class CMakeInfo:
+    executables: t.Mapping[str, ExecutableInfo]
```

Comment:
We should either: (a) rename this to `targets`, and/or (b) maintain two separate mappings (one for binaries and another for libraries)